src/thread_task.c

```diff
@@ -730,14 +731,11 @@ void *dav1d_worker_task(void *data) {
                             dav1d_decode_frame_exit(f, DAV1D_ERR(ENOMEM));
                             f->n_tile_data = 0;
                             pthread_cond_signal(&f->task_thread.cond);
-                            atomic_store(&f->task_thread.init_done, 1);
-                            continue;
```

Comment:
Did you also delete the `continue` from the Rust version?  I don't see a continue there to begin with though.  Did it get turned into a state machine?

Reply:
This is a good question. In the C code, the `continue` statement doesn't seem to make a difference. It is inside a loop `for (int p = 1; p <= 2; p++)`. The code block with `continue` is executed only when `p==2` and there is no statement in the loop after the `continue` statement. Maybe the initial C to Rust translation detected this behaviour and did way with `continue`.

Reply:
Yeah, probably.  The `continue` was deleted anyways, so it should all be good now.